### PR TITLE
override default node selector in suggested namespace

### PR DESCRIPTION
**Pull request by mshitrit**
[ECOPROJECT-1783](https://issues.redhat.com//browse/ECOPROJECT-1783) , [OCPBUGS-22305](https://issues.redhat.com/browse/OCPBUGS-22305)
override the default node selector, in order to make sure that SNR node selector which is defined in the Daemonset will take precedence.

### CODE REVIEW

File: bundle/manifests/self-node-remediation.clusterserviceversion.yaml [hunk 1]
```diff
@@ -41,7 +41,19 @@ metadata:
     description: Self Node Remediation Operator for remediate itself in case of a
       failure.
     olm.skipRange: '>=0.4.0'
-    operatorframework.io/suggested-namespace: openshift-workload-availability
+    operatorframework.io/suggested-namespace-template: |-
```
Comment (slintes) [inline bundle/manifests/self-node-remediation.clusterserviceversion.yaml:44]: we need to keep `operatorframework.io/suggested-namespace`, the template is new OCP 4.13

**Review by slintes — CHANGES_REQUESTED** [review]
Can you test please what happens when the namespace already exists without that annotation? If it isn't added, we should add it to the other operators as well, and need to document it.

**Comment by mshitrit** [comment]
> Can you test please what happens when the namespace already exists without that annotation? If it isn't added, we should add it to the other operators as well, and need to document it.

I've done a quick test:
- created a catsrc with the new version (after I've already created the ns)
- installed snr from that catsrc

As far as I can see the annotation wasn't added to the ns.

However I've also did a simpler test, installing the operator when the ns didn't exist and I saw the same result :sweat_smile: 

since the namesapce was created and the pods where created in the correct ns I'm thinking that only `operatorframework.io/suggested-namespace:` annotation was proccessed from the csv and not `operatorframework.io/suggested-namespace-template:` but it needs furtther investigation.

**Comment by slintes** [comment]
> > Can you test please what happens when the namespace already exists without that annotation? If it isn't added, we should add it to the other operators as well, and need to document it.
> 
> I've done a quick test:
> 
> * created a catsrc with the new version (after I've already created the ns)
> * installed snr from that catsrc
> 
> As far as I can see the annotation wasn't added to the ns.
> 
> However I've also did a simpler test, installing the operator when the ns didn't exist and I saw the same result 😅
> 
> since the namesapce was created and the pods where created in the correct ns I'm thinking that only `operatorframework.io/suggested-namespace:` annotation was proccessed from the csv and not `operatorframework.io/suggested-namespace-template:` but it needs furtther investigation.

just to be sure, you installed using the console in both cases, right? 🙂

**Comment by mshitrit** [comment]
>just to be sure, you installed using the console in both cases, right? 🙂

Indeed I have

**Comment by mshitrit** [comment]
After modifying the format of `operatorframework.io/suggested-namespace-template` I've managed to properly test it.
- In case the namespace does not exit the annotation is added as expected.
- Otherwise it isn't.

I've updated the related epic: https://issues.redhat.com/browse/ECOPROJECT-1761 
@clobrano I see that so far it was only implemented in MDR so make sure to update as well.

**Comment by slintes** [comment]
> * In case the namespace does not exit the annotation is added as expected.
> * Otherwise it isn't.
> 

too bad, thanks for checking!

There seems to be an unrelated code change, otherwise lgtm

File: controllers/selfnoderemediation_controller.go [hunk 2]
```diff
@@ -782,9 +782,9 @@ func (r *SelfNodeRemediationReconciler) updateSnrStatusLastError(snr *v1alpha1.S
 		}
 	}
 
-	_, isUnreconcilableError := err.(*UnreconcilableError)
+	_, isUnrecognisableError := err.(*UnreconcilableError)
```
[the remark this answers is not in the document]
Comment (mshitrit) [reply controllers/selfnoderemediation_controller.go:785]: Just a minor typo fix.

**Pull request merged**